plugins/inspect/repositories_inspector.rb

```diff
@@ -15,93 +15,118 @@
+        "zypper", "--non-interactive", "--xmlout", "repos", "--details", :stdout => :capture
       )
-      credential_files.split("\n").each do |f|
-        content = system.run_command(
-          "cat", "/etc/zypp/credentials.d/#{f}", :stdout => :capture
-        )
-        content.match(/username=(\w*)\npassword=(\w*)/)
-        credentials[f] = {
-          username: $1,
-          password: $2
-        }
+      details = system.run_command(
+        "zypper", "--non-interactive", "repos", "--details", :stdout => :capture
+      ).split("\n").select { |l| l =~ /\A# +\| |\A *\d+ \| / }.
+        map { |l| l.split("|").map(&:strip) }
+    rescue Cheetah::ExecutionFailed => e
+      if e.status.exitstatus == 6
+        description.repositories = RepositoriesScope.new([])
+        return "Found 0 repositories."
+      else
+        raise e
       end
+    end
 
-      reps = Nokogiri::XML(xml).xpath("/stream/repo-list/repo")
-      summary = "Found #{reps.count} repositories."
-      result = reps.map do |rep|
-        pri_value = rep["priority"] ? rep["priority"].to_i : prio[rep["alias"]]
-
-        # NCC
-        rep.at_xpath("./url").text.match(/\?credentials=(\w*)/)
-        cred_value = $1
-        if cred_value && credentials[cred_value]
-          username = credentials[cred_value][:username]
-          password = credentials[cred_value][:password]
-        end
-
-        # SCC
-        rep.at_xpath("./url").text.match(/(https:\/\/updates.suse.com\/SUSE\/)/)
-        scc_url = $1
-        cred_value = "SCCcredentials"
-        if scc_url && credentials[cred_value]
-          username = credentials[cred_value][:username]
-          password = credentials[cred_value][:password]
-        end
-
-        repository = Repository.new(
-          alias:       rep["alias"],
-          name:        rep["name"],
-          type:        rep["type"],
-          url:         rep.at_xpath("./url").text,
-          enabled:     rep["enabled"] == "1",
-          autorefresh: rep["autorefresh"] == "1",
-          gpgcheck:    rep["gpgcheck"] == "1",
-          priority:    pri_value
-        )
-        if username && password
-          repository[:username] = username
-          repository[:password] = password
-        end
-        repository
-      end.sort_by(&:name)
-    else
+    if details.empty?
       result = []
       summary = "Found 0 repositories."
+    else
+      priorities = parse_priorities_from_details(details)
+      credentials = get_credentials_from_system(system)
+      result = parse_repositories_from_xml(xml, priorities, credentials)
+      summary = "Found #{result.count} repositories."
     end
 
     description.repositories = RepositoriesScope.new(result)
     summary
   end
+
+  def parse_priorities_from_details(details)
+    # parse and remove header
+    idx_prio = details.first.index("Priority")
+    idx_alias = details.first.index("Alias")
+    details.shift
+
+    prio = {}
+    details.each_with_index do |entry,idx|
```

Comment:
Unused block argument - `idx`. If it's necessary, use `_` or `_idx` as an argument name to indicate that it won't be used.<br>Space missing after comma.